code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,700 @@
 			var/percent = round((reagents.total_volume / beaker.volume) * 100)
 			switch(percent)
-				if(0 to 9)		filling.icon_state = "reagent0"
-				if(10 to 24) 	filling.icon_state = "reagent10"
-				if(25 to 49)	filling.icon_state = "reagent25"
-				if(50 to 74)	filling.icon_state = "reagent50"
-				if(75 to 79)	filling.icon_state = "reagent75"
-				if(80 to 90)	filling.icon_state = "reagent80"
-				if(91 to INFINITY)	filling.icon_state = "reagent100"
+				if(0 to 9)		filling.icon_state = "[tipped ? "tipped_" : ""]reagent0"
+				if(10 to 24) 	filling.icon_state = "[tipped ? "tipped_" : ""]reagent10"
+				if(25 to 49)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent25"
+				if(50 to 74)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent50"
+				if(75 to 79)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent75"
+				if(80 to 90)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent80"
+				if(91 to INFINITY)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent100"
 
 			var/reagent_color = reagents.get_color()
 			filling.icon += reagent_color
 			add_overlay(filling)
+		if(attached)
+			add_overlay("iv_in[tipped ? "_tipped" : ""]")
+			if(mode)
+				add_overlay("light_green[tipped ? "_tipped" : ""]")
+			else
+				add_overlay("light_red[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		if(istype(tank, /obj/item/tank/oxygen))
+			tank_type = "oxy"
+		else if(istype(tank, /obj/item/tank/anesthetic))
+			tank_type = "anest"
+		else if(istype(tank, /obj/item/tank/phoron))
+			tank_type = "phoron"
+		else
+			tank_type = "other"
+		add_overlay("tank_[tank_type][tipped ? "_tipped" : ""]")
+		update_gauge()
+	if(breath_mask)
+		if(breather)
+			add_overlay("mask_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("mask_off[tipped ? "_tipped" : ""]")
+		if(epp_active)
+			add_overlay("light_blue[tipped ? "_tipped" : ""]")
+
+/obj/machinery/iv_drip/proc/update_gauge()
+	var/gauge_pressure = 0
+	var/last_gauge_pressure
+	if(tank.air_contents)
+		gauge_pressure = tank.air_contents.return_pressure()
+		if(gauge_pressure > 1015) // tanks.dm TANK_IDEAL_PRESSURE 1015
+			gauge_pressure = -1
+		else
+			gauge_pressure = round((gauge_pressure/1015)*tank.gauge_cap)
+	if(gauge_pressure == last_gauge_pressure)
+		return
+	last_gauge_pressure = gauge_pressure
+	add_overlay("[tank.gauge_icon][(gauge_pressure == -1) ? "overload" : gauge_pressure][tipped ? "_tipped" : ""]")
 
-/obj/machinery/iv_drip/MouseDrop(over_object, src_location, over_location)
-	..()
+/obj/machinery/iv_drip/machinery_process()
+	set background = 1
+	if(istype(breather)) // Mask secured
+		if(!tank)
+			return
+		if(breather.species.flags & NO_BREATHE)
+			return
 
-	if(attached)
-		visible_message("[src.attached] is detached from \the [src].")
-		src.attached = null
-		src.update_icon()
-		blood_message_sent = FALSE
-		return
+		if(!breather.Adjacent(src))
+			step_to(src, get_turf(breather), 1)
+			if(prob(20))
```

Comment:
I would prefer a time-since-last-squeaked to make sure this doesn't spam, rather than a prob.

It can have a prob as well if you want I guess.

Reply:
Mm! I thought a prob() would be more convenient since it doesn't need to check for time it has last squeaked, but I'll definitely change it